### PR TITLE
DROOLS-6213: Apply try with resources for file upload

### PR DESCRIPTION
**Pull request by jomarko**
As part of DROOLS-6213 we cleaned up codebase for uploading XLS decision tables and scorecards. During review of drools-wb changes, the same technical debt was found in the appformer abstract class, that is overriden by drools-wb classes, enhanced as part of DROOLS-6213

For more details see:
- https://issues.redhat.com/browse/DROOLS-6213

### Related PR (not an ensemble)
https://github.com/kiegroup/drools-wb/pull/1477

### CODE REVIEW

File: uberfire-services/uberfire-services-backend/src/main/java/org/guvnor/common/services/backend/file/upload/AbstractFileServlet.java [hunk 1]
```diff
@@ -226,30 +226,27 @@ private String processUpload(final FormData item,
...
-                    break;
-                case UPDATE:
+        try (final InputStream fileData = item.getFile().getInputStream()) {
+            final FileOperation operation = item.getOperation();
+            if (operation == FileOperation.CREATE) {
+                doCreate(targetPath,
+                         fileData,
+                         request,
+                         "Uploaded " + getTimestamp());
+            } else if(operation == FileOperation.UPDATE) {
```
Comment (yesamer) [inline uberfire-services/uberfire-services-backend/src/main/java/org/guvnor/common/services/backend/file/upload/AbstractFileServlet.java:243]: @jomarko I would add a ` ` space after `if`

**Review by yesamer — APPROVED** [review]
Excellent @jomarko ! LGTM

File: uberfire-services/uberfire-services-backend/src/main/java/org/guvnor/common/services/backend/file/upload/AbstractFileServlet.java (hunk 1 above)
Comment (jomarko) [reply uberfire-services/uberfire-services-backend/src/main/java/org/guvnor/common/services/backend/file/upload/AbstractFileServlet.java]: yes, well spotted, changes applied, thank you

**Comment by jomarko** [comment]
Jenkins do cdb

**Comment by jomarko** [comment]
@kiegroup/gatekeepers please merge

**Review by tiagobento — APPROVED** [review]

**Comment by tiagobento** [comment]
@jomarko I'm seeing a lot of red checks on this PR, but I'm not sure if this is expected or not, can you please clarify @jomarko?

**Comment by jomarko** [comment]
```
14:22:39  [ERROR] Plugin org.apache.maven.plugins:maven-source-plugin:3.0.1 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.apache.maven.plugins:maven-source-plugin:jar:3.0.1: Could not transfer artifact org.apache.maven.plugins:maven-source-plugin:pom:3.0.1 from/to mirror-central (https://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8443/nexus/content/groups/kie-all/): Transfer failed for https://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8443/nexus/content/groups/kie-all/org/apache/maven/plugins/maven-source-plugin/3.0.1/maven-source-plugin-3.0.1.pom: Connect to bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8443 [bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com/10.0.13.239] failed: Connection refused (Connection refused) -> [Help 1]
```

seems CI is down, lets wait for green checks, sure

**Comment by jomarko** [comment]
Jenkins please retest this

**Comment by jomarko** [comment]
Jenkins do cdb

**Comment by jomarko** [comment]
@tiagobento it is green now

**Pull request merged**